homeassistant/components/weather/smhi.py

```diff
@@ -0,0 +1,247 @@
+"""Swedish weather institute weather sensor.
+
+Implements the sensor for swedhis
+See homeassistant.components.smhi for details
+"""
+
+import asyncio
+import logging
+from datetime import timedelta
+from typing import List
+
+import async_timeout
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_LATITUDE, CONF_LONGITUDE,
+    CONF_NAME, STATE_UNKNOWN, TEMP_CELSIUS)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.util import dt
+
+from homeassistant.components.weather import (
+    WeatherEntity, ATTR_FORECAST_CONDITION, ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TIME, ATTR_WEATHER_WIND_SPEED,
+    ATTR_WEATHER_WIND_BEARING)
+
+from homeassistant.components.smhi.const import (
+    ENTITY_ID_SENSOR_FORMAT)
+
+DEPENDENCIES = ['smhi']
+
+REQUIREMENTS = ['smhi-pkg==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Used to map condition from API results
+CONDITION_CLASSES = {
+    'cloudy': [5, 6],
+    'fog': [7],
+    'hail': [],
+    'lightning': [21],
+    'lightning-rainy': [11],
+    'partlycloudy': [3, 4],
+    'pouring': [10, 20],
+    'rainy': [8, 9, 18, 19],
+    'snowy': [15, 16, 17, 25, 26, 27],
+    'snowy-rainy': [12, 13, 14, 22, 23, 24],
+    'sunny': [1, 2],
+    'windy': [],
+    'windy-variant': [],
+    'exceptional': [],
+}
+
+
+# 5 minutes between retrying connect to API again
+RETRY_TIMEOUT = 5*60
+
+
+async def async_setup_entry(hass: HomeAssistant,
+                            config_entry: ConfigEntry,
+                            config_entries) -> bool:
+    """Add a weather entity from map location."""
+    location = config_entry.data
+    name = location[CONF_NAME]
+
+    entity = SmhiWeather(hass, name, location[CONF_LATITUDE],
+                         location[CONF_LONGITUDE])
+    entity.entity_id = async_generate_entity_id(
+        ENTITY_ID_SENSOR_FORMAT, name, None, hass)
+    entity.start_poll_weather_api()
+
+    config_entries([entity])
+    return True
+
+
+class SmhiWeather(WeatherEntity):
+    """Representation of a weather entity."""
+
+    _forecasts = None
+
+    def __init__(self, hass, name, latitude: str,
+                 longitude: str) -> None:
+        """Initialize the Smhi weather entity."""
+        self._hass = hass
+        self._name = name
+        self._latitude = latitude
+        self._longitude = longitude
+        self._forecasts = None
+
+    def start_poll_weather_api(self):
```

Comment:
You should use the update method instead, you can specify refresh time by defining SCAN_INTERVAL = timedelta(seconds=5)

Reply:
Yes agree, the plan was to use hub to send data to different sensors and weather. Now I shoose to update weather to support cloudiness and no need for extra sensor. Now it makes more sense to use built in. Thanks for suggestion